thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/alert/AlertTaskRunner.java

```diff
@@ -158,11 +159,44 @@ private void runTask() throws Exception {
       return results;
     }
 
-    // Get filtration rule from metric configuration
-    MergedAnomalyResultDTO anyAnomaly = results.get(0);
-    MetricConfigDTO metricConfigDTO =
-        metricConfigDAO.findByMetricAndDataset(anyAnomaly.getMetric(), anyAnomaly.getCollection());
-    Map<String, String> alertFilterInfo = metricConfigDTO.getAlertFilterInfo();
+    // Function ID to Alert Filter
+    Map<Long, AlertFilter> functionAlertFilter = new HashMap<>();
+
+    List<MergedAnomalyResultDTO> qualifiedAnomalies = new ArrayList<>();
+    for (MergedAnomalyResultDTO result : results) {
+      // Lazy initiates alert filter for anomalies of the same anomaly function
+      AnomalyFunctionBean anomalyFunctionSpec = result.getFunction();
+      long functionId = anomalyFunctionSpec.getId();
+      AlertFilter alertFilter = functionAlertFilter.get(functionId);
+      if (alertFilter == null) {
+        // Get filtration rule from anomaly function configuration
+        alertFilter = initiateAlertFilter(anomalyFunctionSpec);
```

Comment:
Should not we store filtration rule at metric level ? 
thats how it is in autoanomaly.